src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationOptions.cs

```diff
@@ -63,46 +63,39 @@ public LinkedInAuthenticationOptions()
-        /// First checks if a preferredLocale is returned from the payload, then try to return the value from it.
-        /// Then checks if a value is available for the current UI culture.
-        /// Finally, returns the first localized value.
+        /// Gets or sets a <c>MultiLocaleString</c> resolver, a function which takes all localized values 
+        /// and an eventual preferred locale from the member and returns the selected localized value.
+        /// The default implementation resolve it in this order:
+        /// 1. Returns the <c>preferredLocale</c> value if it is set and has a value.
+        /// 2. Returns the value corresponding to the <see cref="Thread.CurrentUICulture"/> if it exists.
+        /// 3. Returns the first value.
+        /// </summary>
+        /// <see cref="DefaultMultiLocaleStringResolver(IReadOnlyDictionary{string, string}, string)"/>
+        public Func<IReadOnlyDictionary<string, string>, string, string> MultiLocaleStringResolver { get; set; } = DefaultMultiLocaleStringResolver;
+
+        /// <summary>
+        /// Gets the <c>MultiLocaleString</c> value using the configured resolver.
         /// See <a>https://docs.microsoft.com/en-us/linkedin/shared/references/v2/object-types#multilocalestring</a>
         /// </summary>
         /// <param name="user">The payload returned by the user info endpoint.</param>
         /// <param name="propertyName">The name of the <c>MultiLocaleString</c> property.</param>
         /// <returns>The property value.</returns>
-        private static string GetMultiLocaleString(JObject user, string propertyName)
+        private string GetMultiLocaleString(JObject user, string propertyName)
         {
-            if (user[propertyName] == null)
+            const string localizedKey = "localized";
+            if (user[propertyName] == null || user[propertyName][localizedKey] == null)
             {
                 return null;
             }
 
             var preferredLocale = user[propertyName]["preferredLocale"];
-            const string localizedKey = "localized";
-
-            if (preferredLocale != null)
-            {
-                var preferredKey = $"{preferredLocale["language"]}_{preferredLocale["country"]}";
-                var preferredLocalizedValue = user[propertyName][localizedKey][preferredKey];
-                if (preferredLocalizedValue != null)
-                {
-                    return preferredLocalizedValue.Value<string>();
-                }
-            }
-
-            var currentUiKey = Thread.CurrentThread.CurrentUICulture.ToString().Replace('-', '_');
-            var currentUiLocalizedValue = user[propertyName][localizedKey][currentUiKey];
-            if (currentUiLocalizedValue != null)
-            {
-                return currentUiLocalizedValue.Value<string>();
-            }
+            var preferredLocaleKey = preferredLocale == null ? null : $"{preferredLocale["language"]}_{preferredLocale["country"]}";
```

Comment:
Use `string` rather than `var`.

Reply:
Where is your .editorconfig to enforce contributors to do that? 😉

Reply:
_Should_ have gone in as part of #299 😄

Reply:
I updated my fork with your changes, but I don't get it. In your editorconfig, you have:
```
csharp_style_var_for_built_in_types = true:silent 
csharp_style_var_when_type_is_apparent = true:silent 
csharp_style_var_elsewhere = true:silent 
```
And you asked me to change two `var` into `string`, so you just want to disallow var for built-in types?

Reply:
Sorry, yes, that's what it should be based on how the rest of the code appears to be styled already.

I just used the new VS 2019 16.1 feature to generate an EditorConfig based on existing styles, and I assumed it'd done the right thing because I ran a code cleanup on it after I'd added it.

I didn't fully audit the file though as there as so many settings!